assets/javascripts/modules/checkout/checkout.js

```diff
@@ -0,0 +1,151 @@
+        $EDIT_YOUR_DETAILS = form.$EDIT_YOUR_DETAILS,
+        $EDIT_PAYMENT_DETAILS = form.$EDIT_PAYMENT_DETAILS,
+
+        FIELDSET_COLLAPSED = 'fieldset--collapsed',
+        FIELDSET_COMPLETE = 'data-fieldset-complete',
+        IS_HIDDEN = 'is-hidden';
+
+    var manualAddress = function () {
+        if($MANUAL_ADDRESS.length > 0){
+            bean.on($MANUAL_ADDRESS[0], 'click', function (e) {
+                e.preventDefault();
+                showFullAddressFields();
+                $MANUAL_ADDRESS.addClass(IS_HIDDEN);
+            });
+        }
+    };
+
+    var showFullAddressFields = function () {
+        $FULL_ADDRESS.removeClass(IS_HIDDEN);
+    };
+
+    var toggleFieldsets = function () {
+        if($YOUR_DETAILS_SUBMIT.length > 0){
+            bean.on($YOUR_DETAILS_SUBMIT[0], 'click', function (e) {
+                e.preventDefault();
+                if(validations.validatePersonalDetails()){
+                    emailCheck.warnIfEmailTaken().then(function () {
+                        $FIELDSET_YOUR_DETAILS.addClass(FIELDSET_COLLAPSED).attr(FIELDSET_COMPLETE, '');
+                        $FIELDSET_PAYMENT_DETAILS.removeClass(FIELDSET_COLLAPSED);
+                        $EDIT_YOUR_DETAILS.removeClass(IS_HIDDEN);
+                        $EDIT_PAYMENT_DETAILS.addClass(IS_HIDDEN);
+                    }).catch(function (e) {
+                        console.error("failed email check:", e);
+                    });
+                }
+            });
+        }
+
+        if($PAYMENT_DETAILS_SUBMIT.length > 0){
+            bean.on($PAYMENT_DETAILS_SUBMIT[0], 'click', function (e) {
+                e.preventDefault();
+                if(validations.validatePaymentDetails()){
+                    $FIELDSET_PAYMENT_DETAILS.addClass(FIELDSET_COLLAPSED).attr(FIELDSET_COMPLETE, '');
+                    $FIELDSET_REVIEW.removeClass(FIELDSET_COLLAPSED);
+                    $EDIT_PAYMENT_DETAILS.removeClass(IS_HIDDEN);
+                }
+            });
+        }
+
+        if($EDIT_YOUR_DETAILS.length > 0){
+            bean.on($EDIT_YOUR_DETAILS[0], 'click', function (e) {
+                e.preventDefault();
+                collapseFieldsets();
+                $FIELDSET_YOUR_DETAILS.removeClass(FIELDSET_COLLAPSED);
+                $EDIT_YOUR_DETAILS.addClass(IS_HIDDEN);
+                if ($FIELDSET_PAYMENT_DETAILS.attr(FIELDSET_COMPLETE) !== null) {
+                    $EDIT_PAYMENT_DETAILS.removeClass(IS_HIDDEN);
+                } else {
+                    $EDIT_PAYMENT_DETAILS.addClass(IS_HIDDEN);
+                }
+            });
+        }
+
+        if($EDIT_PAYMENT_DETAILS.length > 0){
+            bean.on($EDIT_PAYMENT_DETAILS[0], 'click', function (e) {
+                e.preventDefault();
+                collapseFieldsets();
+                $FIELDSET_PAYMENT_DETAILS.removeClass(FIELDSET_COLLAPSED);
+                $EDIT_PAYMENT_DETAILS.addClass(IS_HIDDEN);
```

Comment:
Fair enough, this represents the behaviour of existing form workflow and it's not the main subject of this PR. Will consider your suggestion for future refactorings.